src/App.jsx

```diff
@@ -14,45 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const defaultValue = [...goodsFromServer];
+  const [goods, setGoods] = useState(defaultValue);
+  const [sort, setSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  function sortGoods(sortValue) {
+    switch (sortValue) {
+      case 'alphabetically':
+        if (isReversed) {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodB.localeCompare(goodA)));
+        } else {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodA.localeCompare(goodB)));
+        }
+
+        break;
+      case 'length':
+        if (isReversed) {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodB.length - goodA.length));
+        } else {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodA.length - goodB.length));
+        }
+
+        break;
+      default:
+        setGoods(defaultValue);
+    }
+
+    setSort(sortValue);
+  }
+
+  const handleReverse = function reverse() {
+    setGoods([...goods].reverse());
+  };
```

Comment:
You don't need to reverse it directly. Would be better to reverse it in the sort function if `isReversed` equals true